lib/baremetalbasetest.pm

```diff
@@ -0,0 +1,269 @@
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Base module for baremetal test cases
+# Maintainer: Michael Moese <mmoese@suse.de>
+
+package baremetalbasetest;
+
+use base opensusebasetest;
+
+use testapi;
+use strict;
+use warnings;
+
+use HTTP::Tiny;
+use IPC::Run;
+use Socket;
+use Time::HiRes 'sleep';
+
+
+# base class for baremetal tests
+sub new {
+    my ($class, $args) = @_;
+    my $self = $class->SUPER::new($args);
+
+    $token = get_var('BAREMETAL_LOCK_TOKEN', '');
```

Comment:
sorry for my lack of perl knowledge, but isn't the use of token a problem here as it is declared in another sub only (host_unlock)? Especially given that we check for 'BAREMETAL_LOCK_TOKEN' in host_lock().
Again, sorry if I miss something but I pref to ask as this looks fairly suspicious to me. Overall I wonder if it would be easier to have sort of global variable for something like this, or perhaps small structure holding such variables.